lib/core/src/server/preview/custom-webpack-preset.js

```diff
@@ -28,5 +30,15 @@ export async function webpack(config, options) {
   }
 
   logger.info('=> Loading custom webpack config (extending mode).');
-  return mergeConfigs(finalDefaultConfig, customConfig);
+
+  // Restore 4.x behavior, but deprecate this mode of extending webpack
```

Comment:
`webpackFinal` will be called twice again =(

since we are not passing the `baseConfig` to the full control mode function, we can just move the call of the  finalWebpack preset from line 16 into two first conditions (lines 24 and 29)